x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/action/user/TransportChangePasswordActionTests.java

```diff
@@ -169,7 +169,7 @@ public void testIncorrectPasswordHashingAlgorithm() {
         TransportService transportService = new TransportService(Settings.EMPTY, mock(Transport.class), null,
             TransportService.NOOP_TRANSPORT_INTERCEPTOR, x -> null, null, Collections.emptySet());
         Settings passwordHashingSettings = Settings.builder().put(XPackSettings.PASSWORD_HASHING_ALGORITHM.getKey(),
-            randomFrom("pbkdf2_50000", "pbkdf2_100000", "bcrypt11", "bcrypt8", "bcrypt")).build();
+            randomFrom("pbkdf2_50000", "pbkdf2_100000", "bcrypt11", "bcrypt8", "bcrypt4")).build();
```

Comment:
These still can be the same as the one returned by getFastHashingAlgorithmForTest. We need to use a randomFrom in both places with different contents

Reply:
I take the comment back:) still leave this up to Yang

Reply:
Or `randomOtherThan` ? (drive by comment, I haven't checked the underlying logic)

Reply:
I've looked now.
I'd propose:
```
        final String systemHash = randomValueOtherThan(hasher.name(), 
            () -> randomFrom("pbkdf2_50000", "pbkdf2_100000", "bcrypt11", "bcrypt8", "bcrypt"));
        Settings passwordHashingSettings = Settings.builder().put(XPackSettings.PASSWORD_HASHING_ALGORITHM.getKey(), systemHash).build();```
```